pkgs/tools/misc/kamp/package.nix

```
@@ -0,0 +1,39 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  replaceVars,
+  perl,
+  stdenv,
+}:
+
+rustPlatform.buildRustPackage rec {
```

```suggestion
rustPlatform.buildRustPackage (finalAttrs: {
```

Prefer `finalAttrs` over `rec`

---

I got it!

---

I got an exception while adding this. (I did not forgot to close parentheses)
Output is: `error: expected a set but found a function`, I will upload code without this change until I find how to fix. Please don't merge.

---

Thank you for looking at this! I am confusing `buildRustPackage` with `mkDerivation` and think the error is expected and my comment can be ignored - you're correct that other rust packages use `rec` without problem 👏

---

You can generally use `rec` without problem, it's not necessarily an anti-pattern. `finalAttrs` has some nice properties but its advantages are often overstated or unnecessary.

Unless there's really good reason to use it in a specific case, please don't go recommending it as if it's gospel. It's especially confusing to newcomers.

---

Thanks both, I proceed to revolve this conversation.